quaint/src/connector/connection_info.rs

```diff
@@ -298,6 +298,56 @@ impl SqlFamily {
         }
     }
 
+    /// Get the default max rows for a batch insert.
+    pub fn max_insert_rows(&self) -> Option<usize> {
+        match self {
+            #[cfg(feature = "postgresql")]
+            SqlFamily::Postgres => None,
+            #[cfg(feature = "mysql")]
+            SqlFamily::Mysql => None,
+            #[cfg(feature = "sqlite")]
+            SqlFamily::Sqlite => Some(999),
+            #[cfg(feature = "mssql")]
+            SqlFamily::Mssql => Some(1000),
+        }
+    }
+
+    /// Get the max number of bind parameters for a single query, which in targets other
+    /// than Wasm can be controlled with the env var QUERY_BATCH_SIZE.
+    #[cfg(not(target_arch = "wasm32"))]
+    pub fn max_bind_values(&self) -> usize {
+        use once_cell::sync::Lazy;
+
+        static BATCH_SIZE_OVERRIDE: Lazy<Option<usize>> = Lazy::new(|| {
+            std::env::var("QUERY_BATCH_SIZE")
+                .ok()
+                .map(|size| size.parse().expect("QUERY_BATCH_SIZE: not a valid size"))
+        });
+
+        (*BATCH_SIZE_OVERRIDE).unwrap_or(self.default_max_bind_values())
+    }
```

Comment:
fwiw since Rust 1.70 the core functionality of `once_cell` is included in std, although it doesn't provide the `Lazy` convenience wrapper:

```suggestion
    pub fn max_bind_values(&self) -> usize {
        use std::sync::OnceLock;

        static BATCH_SIZE_OVERRIDE: OnceLock<Option<usize>> = OnceLock::new();

        BATCH_SIZE_OVERRIDE
            .get_or_init(|| {
                std::env::var("QUERY_BATCH_SIZE")
                    .ok()
                    .map(|size| size.parse().expect("QUERY_BATCH_SIZE: not a valid size"))
            })
            .unwrap_or(self.default_max_bind_values())
    }
```

Reply:
TIL. Thank you.

Reply:
Fixed in [84ca7a8](https://github.com/prisma/prisma-engines/pull/4656/commits/84ca7a8fb318b40d38ebd270c5ab7b46e2849d2e)